modules/tinymce/src/plugins/help/main/ts/ui/PluginsTab.ts

```diff
@@ -7,13 +7,24 @@ import I18n from 'tinymce/core/api/util/I18n';
 import * as Options from '../api/Options';
 import * as PluginUrls from '../data/PluginUrls';
 
+interface PluginData {
+  // The name is just used for sorting alphabetically.
+  readonly name: string;
+  readonly html: string;
+}
+
 const tab = (editor: Editor): Dialog.TabSpec & { name: string } => {
   const availablePlugins = () => {
     const premiumPlugins = Arr.filter(PluginUrls.urls, ({ type }) => {
       return type === PluginUrls.PluginType.Premium;
     });
-    const premiumPluginList = Arr.map(premiumPlugins, (plugin) => '<li>' + I18n.translate(plugin.name) + '</li>').join('');
 
+    // ASSUMPTION: We want the entries to be sorted POST-translation.
+    const sortedPremiumPlugins = Arr.sort(
```

Comment:
We need to use something more advanced than `[].sort()` if we are doing this post translations.

https://stackoverflow.com/a/61094581/7546 looks like it is available on our supported browsers

Reply:
`Intl.Collator` is the same as `localeCompare` btw: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare. You are right we do need to have a custom comparator function here though.

Reply:
Is using `localCompare` for the sortBy function sufficient?

Reply:
As in: 

```
     const sortedPremiumPlugins = Arr.sort(
-      Arr.map(premiumPlugins, (p) => I18n.translate(p.name))
+      Arr.map(premiumPlugins, (p) => I18n.translate(p.name)),
+      (s1, s2) => s1.localeCompare(s2)
```

Reply:
Or do I need to detect the locale first and pass it through to `localeCompare` ?

Reply:
If so, is `editor.options.getLanguageCode` going to yield the correct answer? 
Or is `tinymce.I18n.getCode` going to be better? That looks like it sets a cell, and it's going to be consistent with what translate is doing. Yeah, that seems safer. I'll try that.

Reply:
From what I'm seeing, we have a singleton for the entire tinymce variable for the current locale for translation. When I try to load editors in different languages, the last one clobbers the first one. From a quick search, I couldn't find a JIRA for this, and talking to QA, they weren't aware it worked like this. This fiddle demonstrates that if you open a German menubar menu, you get Spanish items.

So I guess that I'll use tinymce.I18n.getCode because that is the assumption that everything else seems to be using. `editor.translate` just seems to delegate to `I18n` which is global.

Reply:
You can't use the `I18n.getCode` because the locale you have to pass to `localeCompare` must be a BCP 47 locale code and that's not what's used by TinyMCE's language codes. 

![Uploading image.png…]()

Reply:
Do we have conversion code anywhere? 

I'm happy to say that 9019 **hasn't been** done and just go with basic sorting for now, with a known issue that it doesn't work well for all international languages. It's a step forward to sort at least English, given that we've just been trying to do that manually ? If the code isn't a supported language tag, what happens? Should I identify which language codes aren't using a valid format? 

Thoughts? 

I can revert all sorting code if you'd really prefer, because we aren't handling every case. I disagree with that as an approach (because this is not a step backwards), but I haven't worked on this codebase for a long time, so it's not really my call.

Reply:
Maybe I should just wait until https://ephocks.atlassian.net/browse/TINY-9019?focusedCommentId=120080 is resolved. We might just get able to get rid of the translations.

Reply:
Product has decided that the plugin names shouldn't be translated. Removed translation sorting attempt. Re-requesting review.

Reply:
Did you forget to push the sorting is still there?